src/docfx/lib/markdown/MarkdownEngine.cs

```diff
@@ -319,34 +321,76 @@ private static ErrorBuilder GetErrors()
             return file is null ? default : (_input.ReadString(file).Replace("\r", ""), new SourceInfo(file));
         }
 
-        private string GetImageLink(string path, MarkdownObject origin, string? altText)
+        private string GetImageLink(string path, MarkdownObject origin, string? altText, bool? isIcon)
         {
             if (altText is null && origin is LinkInline linkInline && linkInline.IsImage)
             {
                 altText = ToPlainText(origin);
             }
-
-            return GetImageLink(new SourceInfo<string>(path, origin.GetSourceInfo()), origin, altText, -1);
+            var node = new ImageLinkNode
+            {
+                UrlLink = path,
+                ImageLinkType = isIcon == true ? ImageLinkType.Icon : ImageLinkType.Default,
+                AltText = altText,
+                IsInline = origin.IsInlineImage(-1),
+                SourceInfo = origin.GetSourceInfo(),
+                ParentSourceInfoList = origin.GetInclusionStack(),
+                Monikers = origin.GetZoneLevelMonikers(),
+                ZonePivots = origin.GetZonePivots(),
+                TabbedConceptualHeader = origin.GetTabId(),
+            };
+            return GetLink(new SourceInfo<string>(path, origin.GetSourceInfo()), node);
         }
 
         private string GetImageLink(SourceInfo<string> href, MarkdownObject origin, string? altText, int imageIndex)
         {
-            _contentValidator.ValidateLink(GetRootFilePath(), href, origin, true, altText, imageIndex);
-            var link = GetLink(href, origin);
-            return link;
+            var node = new ImageLinkNode
+            {
+                UrlLink = href,
+                ImageLinkType = ImageLinkType.Default,
+                AltText = altText,
+                IsInline = origin.IsInlineImage(imageIndex),
+                SourceInfo = href.Source,
+                ParentSourceInfoList = origin.GetInclusionStack(),
+                Monikers = origin.GetZoneLevelMonikers(),
+                ZonePivots = origin.GetZonePivots(),
+                TabbedConceptualHeader = origin.GetTabId(),
+            };
+            return GetLink(href, node);
+        }
+
+        private string GetLink(SourceInfo<string> link, MarkdownObject origin)
+        {
+            var node = new HyperLinkNode
+            {
+                UrlLink = link,
+                LinkText = (origin is LinkInline {IsImage: false}) ? ToPlainText(origin) : null,
+                HyperLinkType = origin switch
+                {
+                    AutolinkInline => HyperLinkType.AutoLink,
+                    HtmlBlock or HtmlInline => HyperLinkType.HtmlAnchor,
+                    _ => HyperLinkType.Default,
+                },
+                SourceInfo = link.Source,
+                ParentSourceInfoList = origin.GetInclusionStack(),
+                Monikers = origin.GetZoneLevelMonikers(),
```

Comment:
there are some duplicated code here, can we put linkNode creation logic in ContentValidator, so Markdown Engine doesn't need care too much about validation logic